openstf/feature_engineering/apply_features.py

```diff
@@ -149,185 +55,50 @@ def apply_features(data, minute_list=None, h_ahead=24):
+            if key in timedriven_feature_functions
+        }
+        holiday_feature_functions = {
+            key: holiday_feature_functions[key]
+            for key in feature_set_list
+            if key in holiday_feature_functions
+        }
 
-    for function_group in [lag_functions, timedriven_functions, holiday_functions]:
+    # Add the features to the dataframe using previously defined feature functions
+    for function_group in [
+        lag_feature_functions,
+        timedriven_feature_functions,
+        holiday_feature_functions,
+    ]:
         for name, featfunc in function_group.items():
-            df[name] = df.iloc[:, [0]].apply(featfunc)
-
-    if "windspeed" in list(df):
-        df["windspeed_100mExtrapolated"] = calculate_windspeed_at_hubheight(
-            df.windspeed
-        )
-        df["windPowerFit_extrapolated"] = calculate_windturbine_power_output(
-            df["windspeed_100mExtrapolated"]
-        )
-
-    if "windspeed_100m" in list(df):
-        df["windpowerFit_harm_arome"] = calculate_windturbine_power_output(
-            df["windspeed_100m"].astype(float)
-        )
-
-    # Try to add humidity  calculations, ignore if required columns are missing
-    try:
-        humidity_df = humidity_calculations(df.temp, df.humidity, df.pressure)
-        df = df.join(humidity_df)
-    except AttributeError:
-        pass  # This happens when a required column for humidity_calculations
-        # is not present
-
-    return df
-
-
-def apply_multiple_horizon_features(data, h_aheads=[0.25, 24], latency={"APX": 24}):
-    """Based on ApplyFeatures, but suited for generating a table containing
-    multiple time horizons at once./
-    Also take function latency into consideration
-
-    Input:
-        - data: pd.DataFrame(index = datetime, columns =
-                             [label, predictor_1,..., predictor_n])
-        - hAhead: list. Forecast horizon limit in hours.
-        Features requiring more recent label-data are omitted.
-        - latency: dict(key=value). set data[key] to nan for horizon > value
-
-    Output:
-        - pd.DataFrame(index = datetime, columns =
-            [label, predictor_1,..., predictor_n, feature_1, ..., feature_m])
-
-    Example
-    import pandas as pd
-    import numpy as np
-    index = pd.date_range(start = "2017-01-01 09:00:00",
-                             freq = '15T', periods = 200)
-    data = pd.DataFrame(index = index,
-                        data = dict(load=np.sin(index.hour/24*np.pi)*
-                                    np.random.uniform(0.7,1.7, 200)))"""
-
-    result = pd.DataFrame()
-    cols = []
-    for h_ahead in h_aheads:
-        res = apply_features(data.copy(), h_ahead=h_ahead)
-        res["Horizon"] = h_ahead
-        if len(res.columns) > len(cols):
-            cols = res.columns
-        result = result.append(res, sort=False)  # appending unsorts columns
+            data[name] = data.iloc[:, [0]].apply(featfunc)
```

Comment:
@JanMaartenvanDoorn see my previous comment. If you remove line 76 top 86 you could just check the name here and continue. Als0 I would suggest to remove one level of complexity by removing the outer look.

```python
# CURRENT CODE
    # Only select features that occur in feature_set_list,
    # if feature_set_list is none nothing is removed
    if feature_set_list is not None:
        timedriven_feature_functions = {
            key: timedriven_feature_functions[key]
            for key in feature_set_list
            if key in timedriven_feature_functions
        }
        holiday_feature_functions = {
            key: holiday_feature_functions[key]
            for key in feature_set_list
            if key in holiday_feature_functions
        }

    # Add the features to the dataframe using previously defined feature functions
    for function_group in [
        lag_feature_functions,
        timedriven_feature_functions,
        holiday_feature_functions,
    ]:
        for name, featfunc in function_group.items():
            data[name] = data.iloc[:, [0]].apply(featfunc)
```

```python
# SUGGESTED IMPLEMENTATION
# Add the features to the dataframe using previously defined feature functions
for key, featfunc in feature_functions.items():
    # Don't generate feature is not in feature_set_list
    if key not in feature_set_list:
        continue
    data[key] = data.iloc[:, [0]].apply(featfunc)
```

Reply:
Good one, implemented your suggestion